### PR TITLE
feat: Enable support for large polynomials in VID

### PR DESCRIPTION
**Pull request by akonring**
## Description

This PR introduces the concept of multiplicity of the degree for polynomials encoding the payload.
Support for large degree polynomials is done by extending message/codeword length by a factor $\alpha$ while maintaining the same rate $r=\frac{\alpha\cdot m}{\alpha\cdot n}$. Each node in the system will receive $\alpha$ evaluation points (multi-shares) for each polynomial $f_i$  $\forall i\in \{1,\ldots,k\}$.

closes: #393

### CODE REVIEW

File: primitives/src/vid/advz.rs
Comment (ggutoski) [inline primitives/src/vid/advz.rs:371]: This might be a good use case for [`Vec::drain`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.drain). If that works then you could delete lines 378-379.

File: primitives/src/vid/advz.rs
Comment (ggutoski) [inline primitives/src/vid/advz.rs:528]: As discussed offline, let's strengthen this check to ensure for all shares the evals len equals `common.poly_commits.len()` times `multiplicity`. Then you can know which polynomial belongs to each eval.

File: primitives/src/vid/advz.rs
Comment (ggutoski) [inline primitives/src/vid/advz.rs:758]: ❤️

File: primitives/src/vid/advz.rs
Comment (ggutoski) [inline primitives/src/vid/advz.rs:946]: As discussed, this test is dumb. Please delete it.

**Review by ggutoski — COMMENTED** [review]
Looks great! Only minor changes remain as discussed offline.

**Comment by ggutoski** [comment]
[CI failure](https://github.com/EspressoSystems/jellyfish/actions/runs/7803221583/job/21284984932?pr=480) seems to be the same as this other PR: https://github.com/EspressoSystems/jellyfish/pull/481#issuecomment-1928740539 The fix is the same: wait and try again.

File: primitives/src/vid/advz.rs
Comment (akonring) [reply primitives/src/vid/advz.rs:58]: Yes, went back and forth on this but eventually just left it.
Think my original confusion stemmed from the fact that in coding theory a "chunk" can mean different things (fx see [here](https://en.wikipedia.org/wiki/Erasure_code#Applications_of_erasure_coding_in_storage_systems)).

After grasping the meaning I don't think it is that bad and it is just nice to call `.chunks(chunk_size)` on an iterator :smile:.

Let me know if you have strong opinions on this, then we can change it.

File: primitives/src/vid/advz.rs
Comment (akonring) [reply primitives/src/vid/advz.rs:371]: Clippy is suggesting `std::mem::take(&mut evals)`.
But I'm unsure if introducing this has implications for our WASM compliance?
(Arkworks has a similar `ark_std::mem`)

File: primitives/src/vid/advz.rs
Comment (ggutoski) [reply primitives/src/vid/advz.rs]: On second thought we should keep the test but check that payload recovery returns an error

File: primitives/src/vid/advz.rs
Comment (ggutoski) [reply primitives/src/vid/advz.rs]: No idea. Tempting as it is to learn the answer, let's not get nerd-sniped by this question. I leave it to your judgement. 😁

File: primitives/src/vid/advz.rs
Comment (ggutoski) [inline primitives/src/vid/advz.rs:200]:
```suggestion
    // aggretate_proofs.len() equals self.multiplicity
    // TODO further aggregate into a single KZG proof.
    aggregate_proofs: Vec<KzgProof<E>>,
```
...or some reasonable explainer comment.

**Review by ggutoski — APPROVED** [review]
Just one more suggested code comment.

**Review by ggutoski — APPROVED** [review]
:shipit:

**Review by chancharles92 — COMMENTED** [review]
@akonring Good job!

**Comment by chancharles92** [comment]
@ggutoski @akonring Do we want to add some tests for the case where `multiplicity > 1`?

**Comment by akonring** [comment]
> @ggutoski @akonring Do we want to add some tests for the case where `multiplicity > 1`?

Yes, good point! Should prolly have been a part of this PR but can create another issue and make a separate PR.

**Comment by ggutoski** [comment]
> @ggutoski @akonring Do we want to add some tests for the case where `multiplicity > 1`?

LOL thank you @chancharles92 ! 🤦

**Comment by akonring** [comment]
Issue here: https://github.com/EspressoSystems/jellyfish/issues/483
Thanks for keeping me honest, @chancharles92.

Since I was a little too fast on the merge button, if there are any leftover input/comments we can maybe put them in the new issue #483

**Pull request merged**
